Explorer/Assets/DCL/Roads/Settings/RoadSettingsAsset.cs

```diff
@@ -1,18 +1,137 @@
+                    roadDescription.RoadCoordinate.ParcelToPositionFlat() + ParcelMathHelper.RoadPivotDeviation,
+                    roadDescription.Rotation.SelfOrIdentity(),
+                    Vector3.one);
+
+                ProcessCandidates(prefab.IndirectCandidates, roadRoot, tempIndirectCandidates);
+            }
+
+            IndirectLODGroups = tempIndirectCandidates
+                               .Select(kvp => new GPUInstancingLODGroupWithBuffer(kvp.Key.LODGroup, kvp.Value.ToList()))
+                               .OrderBy(group => group.LODGroup.Name)
+                               .ToList();
+        }
+
+        private void ProcessCandidates(List<GPUInstancingLODGroupWithBuffer> sourceCandidates, Matrix4x4 roadRoot, Dictionary<GPUInstancingLODGroupWithBuffer, HashSet<PerInstanceBuffer>> targetDict)
+        {
+            foreach (GPUInstancingLODGroupWithBuffer candidate in sourceCandidates)
+            {
+                if (!targetDict.TryGetValue(candidate, out HashSet<PerInstanceBuffer> matrices))
+                {
+                    matrices = new HashSet<PerInstanceBuffer>();
+                    targetDict.Add(candidate, matrices);
+                }
+
+                foreach (PerInstanceBuffer instanceData in candidate.InstancesBuffer)
+                    matrices.Add(new PerInstanceBuffer(roadRoot * instanceData.instMatrix));
+            }
+        }
+
+#if UNITY_EDITOR
+        public void CollectGPUInstancingLODGroups(Vector2Int min, Vector2Int max)
+        {
+            Dictionary<string, GPUInstancingPrefabData> loadedPrefabs = LoadAllPrefabs();
+
+            var tempIndirectCandidates = new Dictionary<GPUInstancingLODGroupWithBuffer, HashSet<PerInstanceBuffer>>();
+
+            foreach (RoadDescription roadDescription in RoadDescriptions)
+            {
+                if (IsOutOfRange(roadDescription.RoadCoordinate)) continue;
+
+                if (!loadedPrefabs.TryGetValue(roadDescription.RoadModel, out GPUInstancingPrefabData prefab))
+                {
+                    Debug.LogWarning($"Can't find prefab {roadDescription.RoadModel}, using default");
+                    prefab = loadedPrefabs[RoadAssetsPool.DEFAULT_ROAD_KEY];
+                }
+
+                var roadRoot = Matrix4x4.TRS(
+                    roadDescription.RoadCoordinate.ParcelToPositionFlat() + ParcelMathHelper.RoadPivotDeviation,
+                    roadDescription.Rotation.SelfOrIdentity(),
+                    Vector3.one);
+
+                ProcessCandidates(prefab.IndirectCandidates, roadRoot, tempIndirectCandidates);
+            }
+
+            IndirectLODGroups = tempIndirectCandidates
+                               .Select(kvp => new GPUInstancingLODGroupWithBuffer(kvp.Key.LODGroup, kvp.Value.ToList()))
+                               .OrderBy(group => group.LODGroup.Name)
+                               .ToList();
+
+            EditorUtility.SetDirty(this);
+            AssetDatabase.SaveAssets();
```

Comment:
We at Decentraland use SaveAssets way too much. Please use [SaveAssetIfDirty](https://docs.unity3d.com/2022.3/Documentation/ScriptReference/AssetDatabase.SaveAssetIfDirty.html) instead.

Reply:
Also, please consider supporting correct undo by using Undo.RecordObject instead of EditorUtility.SetDirty. (RecordUndo has to be called *before* you modify the object.)